libkbfs/md_util.go

```diff
@@ -355,19 +356,13 @@ func encryptMDPrivateData(
 			brmd.SetSerializedPrivateMetadata(encodedEncryptedPrivateMetadata)
 		}
 
-		// Sign the writer metadata. This has to be done here,
-		// instead of in signMD, since the MetadataID depends
-		// on it.
-		buf, err := brmd.GetSerializedWriterMetadata(codec)
+		// Sign the writer metadata internally. This has to be
+		// done here, instead of in signMD, since the
+		// MetadataID may depend on it.
+		err := brmd.SignWriterMetadataInternally(ctx, codec, signer)
```

Comment:
Nice, this means we're not going to bother signing journaled mdv3 updates automatically, right?  So switching to mdv3 should be a decent performance win for journaling without needing to do anything extra.